scripts/TRN_TRN.R

```diff
@@ -0,0 +1,264 @@
+  select(trn1, contains("long"))
+
+# Identify unique TRN pairings from pubs fields, fill out boolean information to identify pub source of match
+trn_trn_pubs <- trn_trn_pubs_long |>
+  rowwise() |>
+  mutate(trn2 = list(unique(c(trn_si_long, trn_abs_long, trn_ft_long)) |> na.omit())) |>
+  unnest_longer(trn2) |>
+  group_by(trn1, trn2) |>
+  summarise(pub_si = if_else(any(trn2 == trn_si_long), TRUE, NA),
+            pub_abs = if_else(any(trn2 == trn_abs_long), TRUE, NA),
+            pub_ft = if_else(any(trn2 == trn_ft_long), TRUE, NA)) |>
+  rowwise() |>
+  mutate(trn1inreg2 = trn1 %in% trn_registries$trn2[trn_registries$trn1 == trn2],
+         trn2inreg1 = trn2 %in% trn_registries$trn2[trn_registries$trn1 == trn1]) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, contains("pub")) |>
+  ungroup()
+
+# Integrate pub matches to larger match table
+trn_trn_pubs_tidy <- trn_trn_titles_tidy |>
+  mutate(pub_si = NA, pub_abs = NA, pub_ft = NA) |>
+  rows_upsert(trn_trn_pubs, by = c("trn1", "trn2")) |>
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)) |>
+  select(trn1:trn2inreg1, pub_si, pub_abs, pub_ft,
+         is_match_protocol_sponsor_protocol_id, is_match_results_sponsor_protocol_id,
+         is_title_matched)
+
+
+
+####################################################################################################################
+## Adding so called 'meta booleans' to make it easier to assign priorities to rows for manual validation stage
+
+trn_trn_pubs_tidy <- trn_trn_pubs_tidy |>
+  mutate(at_least_one_EU = if_else((registry1 == "EudraCT" | registry2 == "EudraCT"), TRUE, FALSE),
+         at_least_one_IV = if_else((trn1 %in% trials$id | trn2 %in% trials$id), TRUE, FALSE),
+         at_least_one_pub = if_else(!is.na(pub_si) | !is.na(pub_abs) | !is.na(pub_ft), TRUE, FALSE),
+         at_least_one_sponsor_match = if_else(!is.na(is_match_protocol_sponsor_protocol_id) |
+                                                !is.na(is_match_results_sponsor_protocol_id), TRUE, FALSE))
+
+####################################################################################################################
+# Assigning priorities for manual checks using meta booleans and other columns
+
+# Need to account for NA values here for non 'meta-booleans'
+# Use coalesce() function: returns the first non-NA value in a list. Eg.) coalesce(NA, FALSE) == FALSE ; coalesce(NA, NA, 7) == 7
+
+
+trn_trn_final_tidy <- trn_trn_pubs_tidy |>
+  ungroup() |> # why the ungroup here?
+  mutate(priority = case_when(
+
+    # Priority 1
+    at_least_one_EU &
+      at_least_one_IV &
+      coalesce(trn1inreg2, FALSE) &   # Replace NA with FALSE
+      coalesce(trn2inreg1, FALSE) &   # Replace NA with FALSE
```

Comment:
Delwen said Vladi explained that trn1inreg2 and viceversa should be TRUE/FALSE in the data so please make that change  above. so no na and no need to coalesce